pkgs/applications/misc/keymapp/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, libusb1
+, webkitgtk
+, gtk3
+, writeShellScript
+}:
+
+stdenv.mkDerivation rec {
+  pname = "keymapp";
+  version = "1.0.5";
+
+  src = fetchurl {
+    url = "https://oryx.nyc3.cdn.digitaloceanspaces.com/keymapp/keymapp-${version}.tar.gz";
+    hash = "sha256-e9Ty3gMb+nkXGK8sND4ljyrIxP+1fLasiV6DoTiWmsU=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    libusb1
+    webkitgtk
+    gtk3
+  ];
+
+  sourceRoot = ".";
+
+  installPhase =
+    let
+      bashWrapper = writeShellScript pname ''
+        # This patch is needed when using nvidia GPU https://github.com/NixOS/nixpkgs/issues/266113#issuecomment-1817478988
+        export __NV_PRIME_RENDER_OFFLOAD=1
+        exec @out@/opt/${pname}_bin "$@"
+      '';
+    in
+    ''
+      runHook preInstall
+
+      mkdir -p $out/share/applications
+      mkdir -p $out/opt
+      mkdir -p $out/bin
+
+      install -m755 -D keymapp $out/opt/${pname}_bin
+      substitute ${bashWrapper} $out/bin/${pname} --subst-var out
```

Comment:
Couldn't we use an inline Desktop Item like here: https://github.com/Shawn8901/nix-configuration/blob/ae80dcb4c068c902fdbb7de1c8b76165c9aca599/packages/keymapp/default.nix#L7 
Then we don't have to do the substitution thing (I personally don't like it, if it's preventable)

You could possibly also grab the keymapp.png from my package, so that we have a desktop item. I grabbed that from the AUR package.

Reply:
Thanks for the suggestion. Did it like that :+1: